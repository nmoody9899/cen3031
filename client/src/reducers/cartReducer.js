let initialState = [];

//load cart items from local storage

/*
    Send different types of actions depending on where user is
    switch behavior
*/
if (typeof window !== "undefined") {
  if (localStorage.getItem("cart")) {
    initialState = JSON.parse(localStorage.getItem("cart"));
  } else {
    initialState = [];
  }
}
export function cartReducer(state = initialState, action) {
  switch (action.type) {
    //type payload case TYPE: return PAYLOAD
    //this will be available in state

    case "ADD_TO_CART":
      return action.payload;

    default:
      return state;
  }
}
