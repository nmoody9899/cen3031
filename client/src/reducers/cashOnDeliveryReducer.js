export function cashOnDeliveryReducer(state = false, action) {
  switch (action.type) {
    //type payload case TYPE: return PAYLOAD
    //this will be available in state

    case "CASH_ON_DELIVERY":
      return action.payload;

    default:
      return state;
  }
}
