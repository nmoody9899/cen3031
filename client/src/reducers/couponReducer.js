export function couponReducer(state = false, action) {
  switch (action.type) {
    //type payload case TYPE: return PAYLOAD
    //this will be available in state

    case "COUPON_APPLIED":
      return action.payload;

    default:
      return state;
  }
}
