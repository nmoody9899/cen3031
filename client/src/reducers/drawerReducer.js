export function drawerReducer(state = false, action) {
  switch (action.type) {
    //type payload case TYPE: return PAYLOAD
    //this will be available in state

    case "SET_VISIBLE":
      return action.payload;

    default:
      return state;
  }
}
