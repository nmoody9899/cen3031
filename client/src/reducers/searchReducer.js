/*
    Send different types of actions depending on where user is
    switch behavior
*/
export const searchReducer = (state = { text: "" }, action) => {
  switch (action.type) {
    //type payload case TYPE: return PAYLOAD
    //this will be available in state

    case "SEARCH_QUERY":
      return { ...state, ...action.payload };

    default:
      return state;
  }
};
