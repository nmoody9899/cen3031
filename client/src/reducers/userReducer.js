/*
    Send different types of actions depending on where user is
    switch behavior
*/
 export function userReducer( state = null, action ){
    switch( action.type ){
        //type payload case TYPE: return PAYLOAD
        //this will be available in state

        case "LOGGED_IN_USER": 
            return action.payload;
           
        case "LOGOUT":
            return action.payload; //user: {}    

        default:
            return state;

    }
}