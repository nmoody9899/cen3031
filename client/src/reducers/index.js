//we'll have multiple reducers and redux can help combine them
import { combineReducers } from "redux";

import { userReducer } from "./userReducer";

const rootReducer = combineReducers({
  user: userReducer,
});

export default rootReducer;
