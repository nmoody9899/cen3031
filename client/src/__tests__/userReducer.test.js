import { LogoutOutlined } from "@ant-design/icons";
import configureMockStore from "redux-mock-store";
import thunk from "redux-thunk";
import { userReducer } from "../reducers/userReducer";

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

// allows us to easily return reponses and/or success/fail for a thunk that calls a service
const mockServiceCreator = (body, succeeds = true) => () =>
  new Promise((resolve, reject) => {
    setTimeout(() => (succeeds ? resolve(body) : reject(body)), 10);
  });

describe("userReducer()", () => {
  it("should return the initial state", () => {
    expect(userReducer(undefined, {})).toMatchSnapshot();
  });

  //next test here

  const addRes = () => ({
    //expecting type and payload
    type: "LOGGED_IN_USER",
    payload: {
      name: "user.name",
      email: "user.name@email",
      token: "idTokenResult.tokenstring",
      role: "res.data.role",
      _id: "1",
    },
  });

  it("should dispatch action.payload for LOGGED_IN_USER", () => {
    //initialize mock store with empty state
    const initialState = {};
    const store = mockStore(initialState);
    //dispatch action
    store.dispatch(addRes());

    //
    const actions = store.getActions();
    const expectedPayload = {
      //expecting type and payload
      //type: "LOGGED_IN_USER",
      name: "user.name",
      email: "user.name@email",
      token: "idTokenResult.tokenstring",
      role: "res.data.role",
      _id: "1",
    };
    expect(userReducer(initialState, addRes())).toEqual(expectedPayload);
  });
  //next test here

  it("should dispatch action.payload for LOGOUT", () => {
    //initialize mock store with empty state
    const initialState = {};
    const store = mockStore(initialState);

    const logout = () => ({
      type: "LOGOUT",
      payload: null,
    });

    //dispatch action
    store.dispatch(logout());

    //
    const actions = store.getActions();
    const expectedLogoutPayload = null;
    expect(userReducer(initialState, logout())).toEqual(expectedLogoutPayload);
  });
  //next test under this line
}); //end of describe(userReducer())
