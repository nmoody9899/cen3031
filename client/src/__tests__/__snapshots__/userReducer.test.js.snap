// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`userReducer() should return the initial state 1`] = `null`;
